File: src/main/java/engineer/carrot/warren/thump/util/helper/PlayerHelper.java
package engineer.carrot.warren.thump.util.helper;

import net.minecraft.entity.player.EntityPlayerMP;
import net.minecraft.server.MinecraftServer;
import net.minecraft.util.ChatComponentText;

import java.util.List;

public class PlayerHelper {
    public static List<EntityPlayerMP> getAllPlayers() {
        return MinecraftServer.getServer().getConfigurationManager().playerEntityList;
    }

    public static void sendMessageToAllPlayers(String message) {
        for (EntityPlayerMP player : getAllPlayers()) {
            player.addChatMessage(new ChatComponentText(message));
        }
    }
}

File: src/main/kotlin/engineer/carrot/warren/thump/helper/PlayerHelper.kt
package engineer.carrot.warren.thump.helper

import net.minecraft.entity.player.EntityPlayerMP
import net.minecraft.util.text.ITextComponent
import net.minecraftforge.common.ForgeHooks
import net.minecraftforge.fml.common.FMLCommonHandler

object PlayerHelper {
    val allPlayers: List<EntityPlayerMP>
        get() = FMLCommonHandler.instance().minecraftServerInstance.playerList.players

    fun sendMessageToAllPlayers(message: ITextComponent) {
        for (player in allPlayers) {
            // TODO: Add link finding back in
            player.sendMessage(message)
        }
    }

    @Suppress("UNUSED") fun sendMessageToPlayer(player: String, message: ITextComponent) {
        // TODO: Add link finding back in
        allPlayers.find { it.name == player }?.sendMessage(message)
    }
}
